CIP-1694/README.md

```diff
@@ -0,0 +1,624 @@
+
+#### Other potential acceptance criterion
+
+- [ ] Exercising all the governance actions on a public testnet, with sufficient participation.
+- [ ] Having the constitution text approved by some process.
+- [ ] Having the initial constitutional committee approved by some process.
+
+### Implementation Plan
+
+This proposal restricts itself to the core on-chain mechanism. However, several issues remain to be discussed. These include:
+
+1. off-chain DRep expectations;
+2. bootstrapping the Constitutional Committee;
+3. the initial Constitution;
+4. DRep incentives;
+5. which values in the [ratification requirement table](#requirements) should be hard-coded, and which should be protocol parameters?
+
+Some of these issues may be addressed by changes to this document; others will need to be addressed in new CIPs or discussion documents.
+
+#### New Protocol Parameters
+
+New protocol parameters will be needed for the following:
+
+* the governance action deposit amount
+* governance action expiration (as a number of epochs from the current epoch)
+* thresholds for treasury withdrawals
+* each of the ratification thresholds
+
+As described above, some  of these will be updatable; others will be hard coded.
+
+// TODO: Decide on the initial parameter values and whether they should be updatable.
+
+// TODO: Decide on coherence conditions on the voting thresholds. For example, the threshold for a motion of no-confidence should arguably be higher than that of a minor treasury withdrawal.
+
+In addition, the initial Constitutional Committee and the initial Constitution will need to be defined as part of a bootstrap process.
+
+#### Changes to the existing ledger rules
+
+* The `PPUP` transition rule will be rewritten and moved out of the `UTxO` rule and into the `LEDGER` rule as a new `TALLY` rule.
+
+  It will process the governance actions and the votes, ratify them, and stage governance actions for enactment in the current or next epoch, as appropriate.
+
+* The `NEWEPOCH` transition rule will be modified.
+* The `MIR` sub-rule will be removed.
+* A new `ENACTMENT` rule will be called immediately after the `EPOCH` rule. This rule will enact governance actions that have previously been ratified.
+* The `EPOCH` rule will no longer call the `NEWPP` sub-rule or compute whether the quorum is met on the PPUP state.
+
+#### More on DRep incentives
+
+The DReps arguably need to be compensated for their work.
+
+The corresponding incentive mechanisms need to be specified, with the funds probably coming from the per-epoch treasury allocation.
+Performance constraints will also need to be considered since it would be problematic if millions of DReps were expected to vote on each governance action. Some incentive options to ensure a manageable number of DReps include:
+
+* Requiring a large deposit when registering oneself as a DRep.
```

Comment:
I'm not sure this is a great idea. I think it's a good idea in the case of SPOs, since your stake is directly involved in the method of securing the network, but in the case of dReps, that will be voting on issues regarding the future changes to the network, other factors, such as expertise in a given area is IMO more important.

Reply:
Right. if we want a layer 1 solution, though, we need some way to limit the DReps.

Reply:
maybe eventually, but at the start there could be a lot of interest that is not financially motivated. and we can't just hope that not too many folks want to vote...

Reply:
> maybe eventually, but at the start there could be a lot of interest that is not financially motivated. and we can't just hope that not too many folks want to vote...

I do agree with this. But must the threshold be ADA? As you correctly state, there are no guarantees that this will actually limit the number of representatives, if there are other motives than financial. Maybe if there was a social threshold instead? Or some other kind of proof, like proof of competence. Not sure what that would look like though, I'm just thinking out loud here, and I understand such an alternative solution would come with a large iplementation cost/time asociated with it.

Reply:
Instead of requiring a deposit, we could also require a minimum voting power for a vote be counted. This doesn't block any funds, but it means that most people would have to find someone to delegate to. That would be a bit similar to some governments where parties that don't pass some low percentage hurdle don't get seats at all.

Reply:
Instead of minimum voting power via a threshold, I think another way to do this would be to only count the top N dReps votes by delegated stake. That way the threshold is dynamic based on AVST. And then N could be a protocol paramater that is increased as the blockchain scales.

Although this would almost be functionally equivalent to just picking N random dReps weighted by stake, and that solution would be more inclusive than top N.

Reply:
both of those solutions sound great to me, provided it doesn't leave a lot of people feeling unrepresented.

Reply:
@WhatisRT and @ia-davidpichler both these suggestions are lovely!

> Instead of requiring a deposit, we could also require a minimum voting power for a vote be counted. This doesn't block any funds, but it means that most people would have to find someone to delegate to. That would be a bit similar to some governments where parties that don't pass some low percentage hurdle don't get seats at all.

I think this has some nice attributes, such as simplicity (either you are over the threshold, or not), but how would we handle the case of someone just splitting up to multiple wallets and delegating the minimum amount to get many votes?

> Instead of minimum voting power via a threshold, I think another way to do this would be to only count the top N dReps votes by delegated stake. That way the threshold is dynamic based on AVST. And then N could be a protocol paramater that is increased as the blockchain scales.

This idea is really great to me, as it adds an element of game theory and competetiveness that nicely deals with the concern raised above. I think this suggestion comes with a bit more complexity and might make it a bit harder to understand how your delegation correlates to representation in reality, but I do like the determinism of the votes, and to re-iterate, in my mind this would make it much harder to game the system, as splitting your delegated ADA to multiple walelts could have an adverse effect.

A follow up question: Would the voting power be weighted with regards to ADA or 1 vote = 1 count, per dRep?

> Although this would almost be functionally equivalent to just picking N random dReps weighted by stake, and that solution would be more inclusive than top N.

Even though it might be almost functionality equivalent, I think the determenism and added clarity / predictability and - IMO - perceived honest representation this brings, makes it better in practice.

Reply:
it would have to be weighted by stake (which solves the splitting up problem)

Reply:
A dynamic threshold sounds dangerous: it means that there's the possibility of you being pushed out of the top N and then your votes don't count anymore. You'd also need some tooling that tells you if you are in the top N or not.

Reply:
Yeah, transparency is deceared, I agree with that. If the votes are weighted, that would effectively counteract splitting stakes, even with the hard threshold solution, which was the only real point I had against it.

Reply:
Who Choses N? Should that be one of the items in the constitution? Or should it grow as cardano grows? Should it be a combination of stake pools or producing stake pools and unique active wallets or some other metrics?

Reply:
> Right. if we want a layer 1 solution, though, we need some way to limit the DReps.

Can you highlight what are the limitations to having an arbitrarily large number of DReps on a layer 1?

ADA holders having to register as a DRep with the deposit intended to be large, reduces the freedom of an ADA holder to register their choices themselves (as in a direct democracy). It would be nice if the governance system could bring in the best of both from direct & representative democracy. Ideally, an ADA holder have the fundamental right with **reasonable** restrictions to vote directly or delegate to a DRep. Large deposits can make the restrictions unreasonable.

Reply:
I completely understand your concerns and I feel the same way. Unfortunately, the physical reality today is that if we were to end up with, say, 10x the number of DReps as SPOs, and there were several governance actions to vote on each epoch, this would start to take up a very non-trivial amount of the total throughput of Cardano. This may not happen, but we have to be prepared for this (at least at the beginning when there are so many unknowns).